### PR TITLE
Changes to block names examples

### PR DESCRIPTION
**Pull request by sckott**
AFAICT block names have to be all lowercase, and underscores should be dashes

I ran into this when running the examples, e.g., 

Doesn't relate to any issues that I know of

### Example
```python
from prefect_shell import ShellOperation
ping_op = ShellOperation(commands=["ping -t 1 prefect.io"])
ping_op.save("BLOCK_NAME")
```

```python
ValidationError: 2 validation errors for BlockDocumentCreate
name
  name must only contain lowercase letters, numbers, and dashes (type=value_error)
__root__
  Names must be provided for block documents. (type=value_error)
```

Running versions:
- prefect: 2.7.10
- prefect-shell: 0.1.4

### Checklist

- [ ] References any related issue by including "Closes #<Issue Number>" or "Closes <Issue URL>".
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-shell/issues/new/choose) first.
- [x] Includes tests or only affects documentation.
- [ ] Passes `pre-commit` checks.
  - Run `pre-commit install && pre-commit run --all` locally for formatting and linting.
- [x] Includes screenshots of documentation updates.
  - Run `mkdocs serve` view documentation locally.
- [ ] Summarizes PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-shell/blob/main/CHANGELOG.md) 
    - I wasn't sure you'd want these small changes in the changelog, if accepted, but happy to do so

### CODE REVIEW

**Comment by sckott** [comment]
e.g. of one of the pages changed:

Screen Shot 2023-02-02 at 3 46 57 PM

File: prefect_shell/commands.py [hunk 1]
```diff
@@ -220,7 +220,7 @@ class ShellOperation(JobBlock):
         ```python
         from prefect_shell import ShellOperation
 
-        shell_operation = ShellOperation.load("BLOCK_NAME")
+        shell_operation = ShellOperation.load("block-name")
```
Comment (ahuang11) [inline prefect_shell/commands.py:223]: The UI does a replace on "BLOCK_NAME" here.
```suggestion
        shell_operation = ShellOperation.load("BLOCK_NAME")
```

File: docs/gen_examples_catalog.py [hunk 2]
```diff
@@ -39,7 +39,7 @@ def skip_block_load_code_example(code_example: str) -> bool:
     """
     Skips the code example if it's just showing how to load a Block.
     """
-    return re.search(r'\.load\("BLOCK_NAME"\)\s*$', code_example.rstrip("`"))
+    return re.search(r'\.load\("block-name"\)\s*$', code_example.rstrip("`"))
```
Comment (ahuang11) [inline docs/gen_examples_catalog.py:42]: Needed for UI
```suggestion
    return re.search(r'\.load\("BLOCK_NAME"\)\s*$', code_example.rstrip("`"))
```

**Review by ahuang11 — CHANGES_REQUESTED** [review]
Thanks! I can see how that's confusing; I initially left it with underscore so users are required to change it, but having directly, runnable examples improves user experiences.

The only thing is I would keep are the example and gen_examples_catalog since those are used for UI to find and replace.

**Comment by sckott** [comment]
makes sense, thanks

**Review by ahuang11 — APPROVED** [review]

**Comment by ahuang11** [comment]
Thank you; I'll merge after tests pass.

**Pull request merged**